src/webhook/identity_validation.py

```diff
@@ -0,0 +1,80 @@
+from flask import request, Blueprint
+import sys, traceback
+import os
+from prometheus_client import Counter
+from webhook.request_helper import validate, responses
+
+bp = Blueprint("identity-webhook", __name__)
+
+# define what we track, declare Counter, how many times this route is accessed
+TOTAL_IDENTITY = Counter('webhook_identity_validation_total', 'The total number of identity validation requests')
+DENIED_IDENTITY = Counter('webhook_identity_validation_denied', 'The total number of identity validation requests denied')
+
+identity_prefix = os.getenv("IDENTITY_VALIDATION_PREFIX", "OpenShift_SRE")
+identity_prefix_list = identity_prefix.split(",")
+
+admin_group = os.getenv("GROUP_VALIDATION_ADMIN_GROUP", "osd-sre-admins,osd-sre-cluster-admins")
+admin_groups = admin_group.split(",")
+
+
+@bp.route('/identity-validation', methods=['POST'])
+def handle_request():
+  # inc total identity counter
+  TOTAL_IDENTITY.inc()
+  debug = os.getenv("DEBUG_IDENTITY_VALIDATION", "False")
+  debug = (debug == "True")
+
+  if debug:
+    print("REQUEST BODY => {}".format(request.json))
+
+  valid = True
+  try:
+    valid = validate.validate_request_structure(request.json)
+  except:
+    valid = False
+
+  if not valid:
+    # inc denied identity counter
+    DENIED_IDENTITY.inc()
+    return responses.response_invalid()
+
+  try:
+    body_dict = request.json['request']
+    # If trying to delete a user, must get user name from oldObject instead of object
+    if body_dict['object'] is None:
+      identity_name = body_dict['oldObject']['metadata']['name']
+    else:
+      identity_name = body_dict['object']['metadata']['name']
+    userinfo = body_dict['userInfo']
+    if userinfo['username'] in ("kube:admin", "system:admin", "system:serviceaccount:openshift-authentication:oauth-openshift"):
+      # kube/system admin can do anything
+      if debug:
+        print("Performing action: {} on identity {} by {}".format(body_dict['operation'], identity_name, userinfo['username']))
+      return responses.response_allow(req=body_dict)
+ 
+    sre_user = False
+    for id_prefix in identity_prefix_list:
+      if identity_name.startswith(id_prefix):
```

Comment:
We don't create any identity other than `OpenShift_SRE`, do we need a prefix check or just equals?

Reply:
@jew I thought we may add others for other teams like RHMI and CEE, if we don't, I will remove the list. Here I am using identity name to check the prefix, it has `OpenShift_SRE:Z3NsZWVtYW4` format, I think I should use `providerName` name, so a equals makes more sense.